metpy/calc/thermo.py

```diff
@@ -1482,15 +1482,23 @@ def _find_append_zero_crossings(x, y):
 
     """
     # Find and append crossings to the data
-    crossings = find_intersections(x[1:], y[1:], np.zeros_like(y[1:]) * y.units)
-    x = concatenate((x, crossings[0]))
+    if hasattr(x, 'units'):
+        x_units = x.units
+        x = x.m
+    else:
+        x_units = 1
+    crossings = find_intersections(x[1:], y[1:], np.zeros_like(y[1:]) * y.units, log_x=True)
+    x = concatenate((x, crossings[0] * x_units))
     y = concatenate((y, crossings[1]))
 
     # Resort so that data are in order
     sort_idx = np.argsort(x)
     x = x[sort_idx]
     y = y[sort_idx]
 
+    # Round x values to nearest 7th decimal due to log-conversion
+    x = np.round(x, 7)
```

Comment:
I don't understand what problem this line is solving?

Reply:
So when the conversion back to linear space from log space occurs, it can cause precision issues (i.e coming back with something like 0.99999999 when it really should be 1). This was specifically causing an issue when checking for duplicates in the following lines, which was then failing `test_find_append_zero_crossings` by allowing a value that should have been identified as a duplicate if the conversion was precise. I know this isn't the cleanest thing in the world, so if you have another solution, I'm happy to hear it.

Reply:
Could we modify the calculation of `keep_idx` below to account for precision issues? Maybe just change threshold of 0 to 1e-6?

Reply:
Ah, yup that totally makes more sense and does the same thing. Thanks for your eyes on this!